src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -31,96 +46,1599 @@ const ActorInit En_Elf_InitVars = {
+        phi_f0 = 0.0f;
+    } else {
+        phi_f0 = 100.0f;
+    }
+
+    if (func_8088C804(&this->unk_224, &this->actor.world.pos, phi_f0)) {
+        this->unk_244 = 0;
+        this->unk_248 = 0x200;
+        this->unk_254 = 1.5f;
+        this->unk_26C = func_8088C9CC;
+    } else {
+        this->unk_26C(this, globalCtx);
+    }
+}
+
+void func_8088CC48(EnElf* this, GlobalContext* globalCtx) {
+    EnElf_SetupAction(this, func_8088DD34);
+    this->unk_250 = Rand_ZeroFloat(10.0f) + 10.0f;
+    this->unk_246 = 0;
+    this->unk_24A = (s32)Rand_ZeroFloat(1048.0f) + 0x200;
+    this->unk_224 = this->actor.world.pos;
+    this->unk_258 = randPlusMinusPoint5Scaled(0x7FFF);
+    this->unk_26C = func_8088C9CC;
+    func_8088CBAC(this, globalCtx);
+    this->unk_25C = 0;
+    this->disappearTimer = 240;
+    if ((this->fairyFlags & 0x400) && Actor_GetCollectibleFlag(globalCtx, this->unk_260)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+f32 func_8088CD3C(s32 colorFlag) {
+    switch (colorFlag) {
+        case 1:
+            return Rand_ZeroFloat(55.0f) + 200.0f;
+
+        case 2:
+            return Rand_ZeroFloat(255.0f);
+    }
+
+    return 0.0f;
+}
+
+void EnElf_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnElf* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+    s32 colorConfig;
+    s32 params;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0402AF58, &D_04029140, this->jointTable, this->morphTable, 7);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 15.0f);
+    thisx->shape.shadowAlpha = 255;
+
+    Lights_PointGlowSetInfo(&this->lightInfoGlow, thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, 0xFF,
+                            0xFF, 0xFF, 0);
+    this->lightNodeGlow = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfoGlow);
+    Lights_PointNoGlowSetInfo(&this->lightInfoNoGlow, thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, 0xFF,
+                              0xFF, 0xFF, 0);
+    this->lightNodeNoGlow = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfoNoGlow);
+
+    this->fairyFlags = 0;
+    this->disappearTimer = 600;
+    this->unk_240 = 0.0f;
+    colorConfig = 0;
+
+    this->unk_260 = (this->actor.params & 0xFE00) >> 9;
+    params = this->actor.params & 0xF;
+    if (thisx->params & 0x100) {
+        this->fairyFlags |= 0x400;
+    }
+
+    this->actor.params = params;
+    switch (params) {
+        case 0:
+            thisx->room = -1;
+            EnElf_SetupAction(this, func_8088E850);
+            func_8088C51C(this, 0);
+            this->fairyFlags |= 4;
+            thisx->update = func_8089010C;
+            this->elfMsg = NULL;
+            this->unk_234 = NULL;
+            this->unk_269 = 20;
+            if ((gSaveContext.naviTimer >= 0x64C8) || (gSaveContext.naviTimer < 0xBB8)) {
```

Comment:
```suggestion
            if ((gSaveContext.naviTimer >= 25800) || (gSaveContext.naviTimer < 3000)) {
```
These timers can be in dec